src/main/java/deterministicparrot/DeterministicParrot.java

```diff
@@ -0,0 +1,267 @@
+     *
+     * @param t The task to be added.
+     * @throws Exception If an error occurs while adding the task.
+     */
+    private void addToList(Task t) throws Exception{
+        this.taskList.addTask(t);
+        this.ui.println("     " + "Got it. I've added this task:");
+        this.ui.println("       " + t);
+        this.ui.println("     " + "Now you have " + this.taskList.getSize() + " tasks in the list.");
+        dumpTaskListToFile();
+    }
+    /**
+     * Marks a task as done and updates the UI.
+     *
+     * @param args The arguments passed to the mark command.
+     * @throws Exception If an error occurs while marking the task.
+     */
+    private void markAsDone(String args[]) throws Exception {
+        if(args.length < 2){
+            throw new DeterministicParrotException("Please provide a task number.");
+        }
+        int i;
+        try {
+            i = Integer.parseInt(args[1]);
+        } catch (NumberFormatException e) {
+            throw new DeterministicParrotException("Please provide a valid task number.");
+        }
+        Task t = this.taskList.markAsDone(i);
+        this.ui.println("    " + "Nice! I've marked this task as done:");
+        this.ui.println("       " + t);
+        dumpTaskListToFile();
+    }
+    /**
+     * Marks a task as undone and updates the UI.
+     *
+     * @param toks The arguments passed to the unmark command.
+     * @throws Exception If an error occurs while marking the task as undone.
+     */
+    private void markAsUndone(String toks[]) throws Exception {
+        if(toks.length < 2){
+            throw new DeterministicParrotException("Please provide a task number.");
+        }
+        int i;
+        try {
+            i = Integer.parseInt(toks[1]);
+        } catch (NumberFormatException e) {
+            throw new DeterministicParrotException("Please provide a valid task number.");
+        }
+        Task t = this.taskList.markAsUndone(i);
+        this.ui.println("    " + "OK, I've marked this task as not done yet:\n");
+        this.ui.println("       " + t);
+        dumpTaskListToFile();
+    }
+
+    /**
+     * Adds a "ToDo" task to the list and updates the UI.
+     *
+     * @param args The arguments passed to the todo command.
+     * @throws Exception If an error occurs while adding the task.
+     */
+    private void addToDo(String[] args) throws Exception {
+        if (args.length < 2) {
+            throw new DeterministicParrotException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        String taskDescription = String.join(" ", Arrays.copyOfRange(args, 1, args.length));
+        ToDo t = new ToDo(taskDescription);
+        addToList(t);
+    }
+
+    /**
+     * Adds a "Deadline" task to the list and updates the UI.
+     *
+     * @param args The arguments passed to the deadline command.
+     * @throws Exception If an error occurs while adding the task.
+     */
```

Comment:
Very elegant way of parsing the task + description to deadline & event!😄👍

Reply:
thank you :)